lib/array_equals.rb

```diff
@@ -1,5 +1,26 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
+  i = 0
+
+  if (array1 == nil && array2 == nil) || (array1 == [] && array2 == [])
+    return true
+  elsif array1 == nil || array2 == nil
+    return false
+  end
+
+  if array1.length == array2.length
+    until i > array1.length
+      if array1[i] != array2[i]
+        return false
+      end
+      i += 1
```

Comment:
Tiny nit: typically people use a "for" loop here rather than manually doing `i += 1` (it's more concise, which makes it easier to identify at a glance)